controllers/bucket_controller.go

```diff
@@ -536,28 +570,33 @@ func (r *BucketReconciler) reconcileArtifact(ctx context.Context, obj *sourcev1.
 
 	// Archive directory to storage
 	if err := r.Storage.Archive(&artifact, dir, nil); err != nil {
-		return sreconcile.ResultEmpty, &serror.Event{
+		e := &serror.Event{
 			Err:    fmt.Errorf("unable to archive artifact to storage: %s", err),
 			Reason: sourcev1.StorageOperationFailedReason,
 		}
+		conditions.MarkTrue(obj, sourcev1.StorageOperationFailedCondition,
+			sourcev1.StorageOperationFailedReason, e.Err.Error())
+		return sreconcile.ResultEmpty, e
 	}
-	r.annotatedEventLogf(ctx, obj, map[string]string{
-		"revision": artifact.Revision,
-		"checksum": artifact.Checksum,
-	}, corev1.EventTypeNormal, "NewArtifact", "fetched %d files from '%s'", index.Len(), obj.Spec.BucketName)
 
 	// Record it on the object
 	obj.Status.Artifact = artifact.DeepCopy()
 
 	// Update symlink on a "best effort" basis
 	url, err := r.Storage.Symlink(artifact, "latest.tar.gz")
 	if err != nil {
-		r.eventLogf(ctx, obj, corev1.EventTypeWarning, sourcev1.StorageOperationFailedReason,
-			"failed to update status URL symlink: %s", err)
+		e := &serror.Event{
+			Err:    fmt.Errorf("failed to update status URL symlink: %s", err),
+			Reason: sourcev1.StorageOperationFailedReason,
+		}
+		r.eventLogf(ctx, obj, corev1.EventTypeWarning, sourcev1.StorageOperationFailedReason, e.Err.Error())
+		conditions.MarkTrue(obj, sourcev1.StorageOperationFailedCondition,
```

Comment:
I feel that with the `StorageOperationFailedCondition` being pretty much equal to `StorageOperationFailedReason`, the arbitrary reason could provide a better summary (in all cases where it is set).

Reply:
Addressed by creating granular reasons for all the storage operation failed conditions.